rotkehlchen/chain/ethereum/decoding/kyber/decoder.py

```diff
@@ -0,0 +1,86 @@
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Tuple
+
+from rotkehlchen.accounting.structures import (
+    HistoryBaseEntry,
+    HistoryEventSubType,
+    HistoryEventType,
+)
+from rotkehlchen.chain.ethereum.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.ethereum.decoding.structures import ActionItem
+from rotkehlchen.chain.ethereum.structures import EthereumTxReceiptLog
+from rotkehlchen.chain.ethereum.types import string_to_ethereum_address
+from rotkehlchen.chain.ethereum.utils import asset_normalized_value, ethaddress_to_asset
+from rotkehlchen.types import ChecksumEthAddress, EthereumTransaction
+from rotkehlchen.user_messages import MessagesAggregator
+from rotkehlchen.utils.misc import hex_or_bytes_to_address, hex_or_bytes_to_int
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.ethereum.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.ethereum.manager import EthereumManager
+
+EXECUTE_TRADE = b"\x18I\xbdj\x03\n\x1b\xca(\xb847\xfd=\xe9o='\xa5\xd1r\xfa~\x9cx\xe7\xb6\x14h\x92\x8a9"  # noqa: E501
+
+
+class KyberDecoder(DecoderInterface):
+    def __init__(  # pylint: disable=super-init-not-called
+            self,
+            ethereum_manager: 'EthereumManager',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: MessagesAggregator,
+    ) -> None:
+        self.ethereum_manager = ethereum_manager
+        self.base = base_tools
+        self.msg_aggregator = msg_aggregator
+
+    def _decode_old_swap(  # pylint: disable=no-self-use
+            self,
+            tx_log: EthereumTxReceiptLog,
+            decoded_events: List[HistoryBaseEntry],
+    ) -> Tuple[Optional[HistoryBaseEntry], Optional[ActionItem]]:
+        sender = hex_or_bytes_to_address(tx_log.topics[1])
+        source_token_address = hex_or_bytes_to_address(tx_log.data[:32])
+        destination_token_address = hex_or_bytes_to_address(tx_log.data[32:64])
+
+        source_token = ethaddress_to_asset(source_token_address)
+        if source_token is None:
+            return None, None
+        destination_token = ethaddress_to_asset(destination_token_address)
+        if destination_token is None:
+            return None, None
+
+        spent_amount_raw = hex_or_bytes_to_int(tx_log.data[64:96])
+        return_amount_raw = hex_or_bytes_to_int(tx_log.data[96:128])
+        spent_amount = asset_normalized_value(amount=spent_amount_raw, asset=source_token)
+        return_amount = asset_normalized_value(amount=return_amount_raw, asset=destination_token)
```

Comment:
Why not use the `KyberTrade` event instead? That one has more information which is also needed. Specifically when comparing them it seems to be the only one that has a destination address for the result of the swap.

![2022-02-22_09-25](https://user-images.githubusercontent.com/1658405/155092335-10d5a650-d88f-4997-9b99-db40b16d0c64.png)


 If you also check the contract and read its code you will see that the destination address of the swap is also configurable.

![2022-02-22_09-28](https://user-images.githubusercontent.com/1658405/155092585-4cd2f3d9-f639-4a70-805c-4c7f27418d2f.png)

 So your current solution, using that event is going to miss all cases where the destination of the swap is not the same as the sender.

Reply:
I didn't use that one because when I inspected the topics in the decode action I got only 2
````
[b"\x18I\xbdj\x03\n\x1b\xca(\xb847\xfd=\xe9o='\xa5\xd1r\xfa~\x9cx\xe7\xb6\x14h\x92\x8a9", b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00rw\xf7\x84\x99fBm4]\x8fk\x9a\xfd\x1d=\x89\x180\x83']
````

The first one is the one I used `execute trade` and the other I didn't find in the list

```
>>> bytearray(b"\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00rw\xf7\x84\x99fBm4]\x8fk\x9a\xfd\x1d=\x89\x180\x83").hex()
'0000000000000000000000007277f7849966426d345d8f6b9afd1d3d89183083'
```

Reply:
It's there. If you don't find it you are doing something wrong. The transaction has 8 logs